src/components/InventoryGroupDetail/GroupTabDetails.js

```diff
@@ -24,12 +25,44 @@ const GroupTabDetailsWrapper = ({
   hasEdgeImages,
 }) => {
   const [tab, setTab] = useState(0);
+
   const { hasAccess: canViewHosts } = usePermissionsWithContext(
     REQUIRED_PERMISSIONS_TO_READ_GROUP_HOSTS(groupId)
   );
+  const conventionalSystemsContent = (
+    <GroupSystems
+      groupName={groupName}
+      groupId={groupId}
+      hostType={hybridInventoryTabKeys.conventional.key}
+    />
+  );
+
+  const immutableSystemsContent = (
+    <GroupImmutableSystems
+      groupId={groupId}
+      groupName={groupName}
+      hostType={hybridInventoryTabKeys.immutable.key}
+    />
+  );
```

Comment:
I think these components have to be memoised with useMemo and dependency array set to [groupId, groupName]. Otherwise, the contents are re-calculated with each re-render.